ui/v2.5/src/models/list-filter/criteria/tags.ts

```diff
@@ -1,51 +1,70 @@
 import { CriterionModifier } from "src/core/generated-graphql";
-import { CriterionType } from "../types";
 import { CriterionOption, IHierarchicalLabeledIdCriterion } from "./criterion";
 
 export class TagsCriterion extends IHierarchicalLabeledIdCriterion {}
 
-class tagsCriterionOption extends CriterionOption {
-  constructor(messageID: string, value: CriterionType, parameterName: string) {
-    const modifierOptions = [
-      CriterionModifier.Includes,
-      CriterionModifier.IncludesAll,
-      CriterionModifier.Equals,
-    ];
-
-    let defaultModifier = CriterionModifier.IncludesAll;
-
-    super({
-      messageID,
-      type: value,
-      parameterName,
-      modifierOptions,
-      defaultModifier,
-    });
-  }
-}
-
-export const TagsCriterionOption = new tagsCriterionOption(
-  "tags",
-  "tags",
-  "tags"
-);
-export const SceneTagsCriterionOption = new tagsCriterionOption(
-  "sceneTags",
-  "sceneTags",
-  "scene_tags"
-);
-export const PerformerTagsCriterionOption = new tagsCriterionOption(
-  "performerTags",
-  "performerTags",
-  "performer_tags"
-);
-export const ParentTagsCriterionOption = new tagsCriterionOption(
-  "parent_tags",
-  "parentTags",
-  "parents"
-);
-export const ChildTagsCriterionOption = new tagsCriterionOption(
-  "sub_tags",
-  "childTags",
-  "children"
-);
+export const TagsCriterionOption = new CriterionOption({
+  messageID: "tags",
+  type: "tags",
+  parameterName: "tags",
+  modifierOptions: [
+    CriterionModifier.IncludesAll,
+    CriterionModifier.Includes,
+    CriterionModifier.Equals,
+    CriterionModifier.IsNull,
+    CriterionModifier.NotNull,
+  ],
```

Comment:
Since these are all the same, make a local list value and use that instead.

Reply:
Actually these aren't quite the same, but I'm not sure why.

Reply:
I went through each one to check what modifiers actually worked, and used which worked for each one. Excludes is in all but the first one because the excluded option wasn't working for them, I'll quickly test #3775 and then they should be able to be removed. Equals is only present for the normal tags modifier because it also didn't seem to work for the others.

Ideally they should indeed all be the same, when equals/not equals works for all the criteria.